File: LeetCodeRecord/MethodFirst/617-MergeTrees.java
package MethodFirst;

class MergeTrees {
    class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;

        TreeNode (int x) {
            val = x;
        }
    }

    public TreeNode mergeTrees(TreeNode t1, TreeNode t2) {
        if (t1 == null && t2 == null) {
            return null;
        }

        if (t1 != null && t2 == null) {
            return t1;
        }

        if (t1 == null && t2 != null) {
            return t2;
        }

        TreeNode left = mergeTrees(t1.left, t2.left);
        TreeNode right=  mergeTrees(t1.right, t2.right);

        t1.val = t1.val + t2.val;
        t1.left = left;
        t1.right = right;

        return t1;
    }
}

File: Top200/617-MergeTrees.go
package main

/**
 * Definition for a binary tree node.
 * type TreeNode struct {
 *     Val int
 *     Left *TreeNode
 *     Right *TreeNode
 * }
 */

type TreeNode struct {
	Val   int
	Left  *TreeNode
	Right *TreeNode
}

func mergeTrees(root1 *TreeNode, root2 *TreeNode) *TreeNode {
	if root1 == nil && root2 == nil {
		return nil
	}

	if root1 != nil && root2 == nil {
		return root1
	}

	if root1 == nil && root2 != nil {
		return root2
	}

	root1.Val = root1.Val + root2.Val

	left := mergeTrees(root1.Left, root2.Left)
	right := mergeTrees(root1.Right, root2.Right)

	root1.Left = left
	root1.Right = right

	return root1
}
